pkgs/servers/scaphandre/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv
+, lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, powercap
+, runCommand
+, dieHook
+}:
+
+let
+  scaphandre = rustPlatform.buildRustPackage rec {
+    pname = "scaphandre";
+    version = "0.5.0";
+
+    src = fetchFromGitHub {
+      owner = "hubblo-org";
+      repo = pname;
+      rev = "v${version}";
+      sha256 = "sha256-cXwgPYTgom4KrL/PH53Fk6ChtALuMYyJ/oTrUKHCrzE=";
+    };
+
+    cargoSha256 = "sha256-Vdkq9ShbHWepvIgHPjhKY+LmhjS+Pl84QelgEpen7Qs=";
+
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ openssl ];
+
+    checkPhase = ''
+      runHook preCheck
+
+      # Work around to pass test due to non existing path
+      # https://github.com/hubblo-org/scaphandre/blob/v0.5.0/src/sensors/powercap_rapl.rs#L29
+      export SCAPHANDRE_POWERCAP_PATH="/tmp/scaphandre"
+
+      mkdir -p "$SCAPHANDRE_POWERCAP_PATH"
+
+      runHook postCheck
+    '';
+
+    passthru.tests = {
+      stdout = runCommand "${pname}-test" {
+        buildInputs = [
+          scaphandre
+          dieHook
+        ];
+      } ''
+        ${scaphandre}/bin/scaphandre stdout -t 4 > $out  || die "Scaphandre failed to measure consumption"
```

Comment:
didn't you say you need intel_rapl_common? how is it supposed to work on non-Intel machines?

Reply:
Yes it does, I thought that when building the package it was going through the tests. Should I extract it into a nixosTest ? According to the [compatibility](https://hubblo-org.github.io/scaphandre-documentation/compatibility.html) page, the sensor is available on Intel & AMD but I don't know how to handle that.

Reply:
I added a nixosTest to ensure kernel module `intel_rapl_common` is activated.